### PR TITLE
Handle publishing a different tag than what is currently checked out

### PR DESCRIPTION
**Pull request by davisagli**
- This gives the metadeploy publishing task the ability to release a tag even if it's not what's currently checked out. The task fetches the archive for the specified tag into a temporary directory, constructs a project_config for it, and uses that to freeze the installation steps.
- Make get_static_dependencies return a specific commit ref for metadata bundles -- for the tag if specified, otherwise for the most recent release if there is one, otherwise for the head of master.

# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

File: cumulusci/core/config/BaseProjectConfig.py [hunk 1]
```diff
@@ -546,16 +546,37 @@ def process_github_dependency(self, dependency, indent=None, include_beta=None):
...
+                    release = repo.release_from_tag(dependency["tag"])
+                except NotFoundError:
+                    raise DependencyResolutionError(
+                        "{}No release found for tag {}".format(
+                            indent, dependency["tag"]
+                        )
+                    )
+            else:
+                release = self._find_latest_release(repo, include_beta)
+            if release:
```
Comment (cdcarter) [inline cumulusci/core/config/BaseProjectConfig.py]: add some vertical space here

File: cumulusci/core/config/BaseProjectConfig.py [hunk 2]
```diff
@@ -546,16 +546,37 @@ def process_github_dependency(self, dependency, indent=None, include_beta=None):
...
+                ref = repo.tag(
+                    repo.ref("tags/" + release.tag_name).object.sha
+                ).object.sha
```
Comment (cdcarter) [inline cumulusci/core/config/BaseProjectConfig.py]: `release.target_commitish` ?

File: cumulusci/core/utils.py [hunk 3]
```diff
@@ -130,7 +131,7 @@ def dictmerge(a, b, name=None):
                     if key in a:
                         a[key] = dictmerge(a[key], b[key], name)
                     else:
-                        a[key] = b[key]
+                        a[key] = copy.copy(b[key])
```
Comment (cdcarter) [inline cumulusci/core/utils.py]: oh nice

File: cumulusci/tasks/metadeploy.py [hunk 4]
```diff
@@ -67,12 +71,38 @@ class Publish(BaseMetaDeployTask):
...
+                repo_info={
+                    "root": project_dir,
+                    "owner": repo_owner,
+                    "name": repo_name,
+                    "url": self.project_config.repo_url,
+                    "branch": tag,
+                    "commit": commit_sha,
+                },
+            )
+            project_config.set_keychain(self.project_config.keychain)
```
Comment (cdcarter) [inline cumulusci/tasks/metadeploy.py]: not sure if you actually need this, but i guess, cant hurt

File: cumulusci/core/config/BaseProjectConfig.py (hunk 2 above)
Comment (davisagli) [reply cumulusci/core/config/BaseProjectConfig.py]: I hoped so too, but that returns "master"

**Pull request merged**